app/models/user.rb

```diff
@@ -12,13 +12,16 @@ class User < ApplicationRecord
   has_many :uncompleted_needs, -> { uncompleted }, class_name: 'Need'
   has_many :uncompleted_contacts, through: :uncompleted_needs, source: :contact
   has_many :completed_contacts, through: :completed_needs, source: :contact
+  has_many :user_roles, dependent: :destroy
+  has_many :roles, through: :user_roles
+  belongs_to :role, optional: true
```

Comment:
I'm not sure that the role should be optional.

Reply:
This links in with the data migration issue, e.g. need to add this field, and also assign to a role at the same time.